src/documents/CoreAPI.md

```diff
@@ -0,0 +1,635 @@
+
+* `array {Array}` The array to be flattened
+* Returns: `{Array}` The flattened array.
+
+
+## Sorting and searching
+
+### fluid.find(source, func[, deflt])
+
+Scan through a list or hash of objects, terminating on the first member which
+matches a predicate function. The return is the return value from the predicate.
+* `source {Arrayable|Object}` The list or hash of objects to be searched.
+* `func {Function: (element {Any}, index {Number|String}) → Any}` A predicate function, acting on a member. A predicate which
+returns any value which is not `undefined` will terminate the search.
+* `deflt {Object|Undefined}` [optional] A value to be returned in the case no predicate function matches
+a member. The default will be the natural value of `undefined`
+* Returns: The first return value from the predicate function which is not `undefined`
+
+### fluid.find_if(source, func[, deflt])
+
+Scan through a list or hash of objects, terminating on the first member which
+matches a predicate function. The return value is the first element itself for which the predicate returns other than `false`.
+* `source {Arrayable|Object}` The list or hash of objects to be searched.
+* `func {Function: (element {Any}, index {Number}) → Boolean}` A predicate function, acting on a member. A predicate which
+returns any value which is not `false` will terminate the search.
+* `deflt {Object}` [optional] A value to be returned in the case no predicate function matches a
+member. The default will be the natural value of `undefined`
+* Returns: The first element for which the value of the predicate function is not `false`
+
+### fluid.contains(tosearch, value)
+
+Searches through the supplied structure, and returns `true` if the supplied value can be found, using `===` equality
+
+* `tosearch {Object|Array}` The structure to be searched for the supplied value,
+* `value {Any}` The value to be searched for
+* Returns: `{Boolean|Undefined}` Returns `true` if the value was found in the container
+
+### fluid.keyForValue(tosearch, value) {
+
+Searches through the supplied object for the first value which matches the one supplied.
+
+* `tosearch {Object|Array}` The structure to be searched for the supplied value
+* `value {Any}` The value to be found. This will be compared against the structure's members using `===` equality.
+* Returns: `{String|Integer|Undefined}` The first key whose value matches the one supplied, or `undefined` is no value matches
+
+### fluid.stableSort(array, func) 
+
+Applies a stable sorting algorithm to the supplied array and comparator (note that [Array.sort](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort) in JavaScript is not specified
+to be stable). The algorithm used will be an insertion sort, which whilst quadratic in time, will perform well
+on small array sizes.
+* `array {Array of Any-Element}` The array to be sorted. This input array will be modified in place.
```

Comment:
The phrase "this input array will be modified in place" should probably be bolded, the way you have with other similar warnings